### ISSUE TITLE
Accuracy Metric does not seem to work correctly

### ISSUE THREAD

**Issue by sbrunk** [eaplatanios/tensorflow_scala#61]
Running the MNIST example the accuracy metric hook produces odd results. Note that the numbers don't change at all between step 1000 and 10000.

```
2017-11-25 20:11:14.660 [main] INFO  Learn / Hooks / Evaluation - Computing Train Evaluation.
2017-11-25 20:11:16.151 [main] INFO  Learn / Hooks / Evaluation - Step 0 Train Evaluation:
2017-11-25 20:11:16.151 [main] INFO  Learn / Hooks / Evaluation - 	  Accuracy:     0.0974
2017-11-25 20:11:16.238 [main] INFO  Learn / Hooks / Evaluation - Computing Test Evaluation.
2017-11-25 20:11:16.791 [main] INFO  Learn / Hooks / Evaluation - Step 1 Test Evaluation:
2017-11-25 20:11:16.791 [main] INFO  Learn / Hooks / Evaluation - 	  Accuracy:     0.0912
2017-11-25 20:11:17.314 [main] INFO  Learn / Hooks / Evaluation - Computing Test Evaluation.
2017-11-25 20:11:17.905 [main] INFO  Learn / Hooks / Evaluation - Step 0 Test Evaluation:
2017-11-25 20:11:17.906 [main] INFO  Learn / Hooks / Evaluation - 	  Accuracy:     0.2954

...

2017-11-25 20:11:22.399 [main] INFO  Learn / Hooks / Evaluation - Computing Train Evaluation.
2017-11-25 20:11:23.626 [main] INFO  Learn / Hooks / Evaluation - Step 1000 Train Evaluation:
2017-11-25 20:11:23.627 [main] INFO  Learn / Hooks / Evaluation - 	  Accuracy:     0.2918
2017-11-25 20:11:23.627 [main] INFO  Learn / Hooks / Evaluation - Computing Test Evaluation.
2017-11-25 20:11:24.139 [main] INFO  Learn / Hooks / Evaluation - Step 1000 Test Evaluation:
2017-11-25 20:11:24.140 [main] INFO  Learn / Hooks / Evaluation - 	  Accuracy:     0.2954

...

2017-11-25 20:12:13.016 [main] INFO  Learn / Hooks / Evaluation - Computing Train Evaluation.
2017-11-25 20:12:14.321 [main] INFO  Learn / Hooks / Evaluation - Step 10000 Train Evaluation:
2017-11-25 20:12:14.321 [main] INFO  Learn / Hooks / Evaluation - 	  Accuracy:     0.2918
2017-11-25 20:12:14.322 [main] INFO  Learn / Hooks / Evaluation - Computing Test Evaluation.
2017-11-25 20:12:14.819 [main] INFO  Learn / Hooks / Evaluation - Step 10000 Test Evaluation:
2017-11-25 20:12:14.819 [main] INFO  Learn / Hooks / Evaluation - 	  Accuracy:     0.2954
2017-11-25 20:12:14.819 [main] INFO  Learn / Hooks / Termination - Stop requested: Exceeded maximum number of steps.
2017-11-25 20:12:15.956 [main] INFO  Examples / MNIST - Train accuracy = 0.93341666
2017-11-25 20:12:16.271 [main] INFO  Examples / MNIST - Test accuracy = 0.9157
```

**Comment by eaplatanios** [comment]
@sbrunk This is expected actually. The evaluation hook loads the latest saved checkpoint and performs evaluation. This means that if you don't save checkpoints in-between you should not expect the evaluation result to change. I'm not sure if there is something better to do than this given that the evaluation needs to be performed in a new session, but I'm still working on this hook so it might change in the future.

**Comment by eaplatanios** [comment]
Can you confirm that things work as expected if you save checkpoints more regularly so we can close this? Also, please let me know if you have suggestions about what else to do about the evaluation hook and in general if you have any feedback about the learn API (because I remember you were planning to work on a similar high-level API at some point). :)

**Comment by sbrunk** [comment]
Yes I can confirm that it works fine for me when changing the checkpoint saving hook trigger to match the one for evaluation. Thanks!

Regarding feedback: I've just started to look into the examples in more detail and also to create a few on my own to get to know the learn API better. So I'm probably going to pester you with a lot of questions. :)

A few remarks so far:

* The layer composition API looks really nice. I'll try to give you more feedback about it once I've digged a bit deeper.
* The `Model` creation parameters were a bit confusing for me (`input` vs. `trainInput` vs. `trainingInputLayer`). I guess this will become clearer once `Model` and `Input` have more docs but perhaps the naming could be improved, too.
* From a usability perspective, I guess it makes sense to think about adding even more high-level building blocks (configurable, but with reasonable defaults) because I think that could be really great for doing rapid prototyping. I mean not only at the layer level but also i.e. having a reasonable set of preconfigured hooks available as `defaultHooks` or something like that.
